test/unit/Transaction.js

```diff
@@ -317,167 +318,58 @@ describe("Transaction", function () {
+                .setKey(account.publicKey)
+                .setNodeAccountIds(nodeAccountIds)
+                .setTransactionId(txId)
+                .freeze();
         });
 
-        it("should add multiple signatures corresponding to each transaction", function () {
-            const mockedSignatures = [
-                new Uint8Array([10, 11, 12]),
-                new Uint8Array([13, 14, 15]),
-            ];
-
-            transaction._signedTransactions.length = 2;
-            transaction._signedTransactions.list = [
-                {
-                    bodyBytes: new Uint8Array([1, 2, 3]),
-                    sigMap: { sigPair: [] },
-                },
-                {
-                    bodyBytes: new Uint8Array([4, 5, 6]),
-                    sigMap: { sigPair: [] },
-                },
-            ];
+        it("should add a single signature when one transaction is present", async function () {
+            const sigMap = new SignatureMap();
+            const pubKey = account.publicKey;
+
+            for (const tx of transaction._signedTransactions.list) {
+                //                const sig = op.sign(txChunk.bodyBytes);
+                const txBody = HashgraphProto.proto.TransactionBody.decode(
+                    tx.bodyBytes,
+                );
+                const txId = TransactionId._fromProtobuf(txBody.transactionID);
+                const nodeAccountId = AccountId._fromProtobuf(
+                    txBody.nodeAccountID,
+                );
+                const sig = account.sign(tx.bodyBytes);
+                sigMap.addSignature(nodeAccountId, txId, pubKey, sig);
+            }
 
-            // Update stub to return correct signatures for each call
-            mockedPublicKey._toProtobufSignature
-                .onCall(0)
-                .returns({ ed25519: mockedSignatures[0] })
-                .onCall(1)
-                .returns({ ed25519: mockedSignatures[1] });
+            transaction.addSignature(pubKey, sigMap);
 
-            transaction.addSignature(mockedPublicKey, mockedSignatures);
+            const signatures = transaction
+                .getSignatures()
+                .getFlatSignatureList();
 
-            // Verify the signatures were added correctly
-            expect(
-                transaction._signedTransactions.get(0).sigMap.sigPair,
-            ).to.deep.equal([{ ed25519: mockedSignatures[0] }]);
-            expect(
-                transaction._signedTransactions.get(1).sigMap.sigPair,
-            ).to.deep.equal([{ ed25519: mockedSignatures[1] }]);
-
-            // Validate the calls to _toProtobufSignature
-            sinon.assert.calledWith(
-                mockedPublicKey._toProtobufSignature,
-                mockedSignatures[0],
-            );
-            sinon.assert.calledWith(
-                mockedPublicKey._toProtobufSignature,
-                mockedSignatures[1],
+            expect(signatures.length).to.equal(
```

Comment:
Could you please check if the signatures are the same as well. 

*this is valid for all the signature related tests